### PR TITLE
Improve map element_at performance by utilizing MapBlock O(1) access

### PR DESCRIPTION
**Pull request by haozhun**

### CODE REVIEW

File: presto-main/src/test/java/com/facebook/presto/type/TestMapOperators.java [hunk 1]
```diff
@@ -442,22 +442,35 @@ public void testJsonToMap()
     public void testElementAt()
             throws Exception
     {
+        // empty map
```
Comment (martint) [inline presto-main/src/test/java/com/facebook/presto/type/TestMapOperators.java]: Why did the test cases have to change?

Comment (haozhun) [inline presto-main/src/test/java/com/facebook/presto/type/TestMapOperators.java]: The original tests didn't cover reasonable corner cases. It's also not grouped in a way that helps the reader tell what's covered.

Comment (martint) [inline presto-main/src/test/java/com/facebook/presto/type/TestMapOperators.java]: Is every case that existed before covered by the new tests? I looked briefly and it seemed that some of them have been removed (?)

Comment (haozhun) [inline presto-main/src/test/java/com/facebook/presto/type/TestMapOperators.java]: One is removed:

```
assertFunction("element_at(MAP(ARRAY [1], ARRAY [1]), 2)", INTEGER, null);
```

But I have these 5:
```
// missing key
assertFunction("element_at(MAP(ARRAY [1], ARRAY [1.0]), 2)", DOUBLE, null);
assertFunction("element_at(MAP(ARRAY [1.0], ARRAY ['a']), 2.0)", createVarcharType(1), null);
assertFunction("element_at(MAP(ARRAY ['a'], ARRAY [true]), 'b')", BOOLEAN, null);
assertFunction("element_at(MAP(ARRAY [true], ARRAY [ARRAY [1]]), false)", new ArrayType(INTEGER), null);
assertFunction("element_at(MAP(ARRAY [ARRAY [1]], ARRAY [1]), ARRAY [2])", INTEGER, null);
```

The rest may be re-formatted, but are all still there.

**Pull request merged**